File: fb_alarabiya_bot.php
<?php
/********************
 * الإعدادات
 ********************/
$facebookPage = "https://mbasic.facebook.com/AlArabiya";
$telegramBotToken = "PUT_BOT_TOKEN_HERE";
$telegramChatId   = "PUT_CHAT_ID_HERE";

$hashFile = __DIR__ . "/last_post_hash.txt";

/********************
 * دوال مساعدة
 ********************/
function fetchPage($url) {
    $ch = curl_init($url);
    curl_setopt_array($ch, [
        CURLOPT_RETURNTRANSFER => true,
        CURLOPT_FOLLOWLOCATION => true,
        CURLOPT_USERAGENT => "Mozilla/5.0",
        CURLOPT_TIMEOUT => 20
    ]);
    $html = curl_exec($ch);
    curl_close($ch);
    return $html;
}

function sendTelegram($token, $chatId, $message) {
    $url = "https://api.telegram.org/bot{$token}/sendMessage";
    $data = [
        "chat_id" => $chatId,
        "text" => $message,
        "disable_web_page_preview" => false
    ];
    $ch = curl_init($url);
    curl_setopt_array($ch, [
        CURLOPT_POST => true,
        CURLOPT_POSTFIELDS => $data,
        CURLOPT_RETURNTRANSFER => true
    ]);
    curl_exec($ch);
    curl_close($ch);
}

function getLastHash($file) {
    return file_exists($file) ? trim(file_get_contents($file)) : "";
}

function saveHash($file, $hash) {
    file_put_contents($file, $hash);
}

/********************
 * جلب آخر منشور
 ********************/
$html = fetchPage($facebookPage);

if (!$html) {
    sendTelegram($telegramBotToken, $telegramChatId, "⚠️ فشل جلب صفحة فيسبوك");
    exit;
}

/*
 نبحث عن أول رابط منشور
 mbasic يعرض الروابط بهذا الشكل تقريبًا:
 /story.php?story_fbid=XXXX&id=YYYY
*/
if (!preg_match('/\/story\.php\?story_fbid=([0-9]+)&id=([0-9]+)/', $html, $matches)) {
    sendTelegram($telegramBotToken, $telegramChatId, "⚠️ لم يتم العثور على منشور");
    exit;
}

$postUrl = "https://www.facebook.com/story.php?story_fbid={$matches[1]}&id={$matches[2]}";

/********************
 * جلب صفحة المنشور نفسه
 ********************/
$postHtml = fetchPage("https://mbasic.facebook.com" . $matches[0]);

$text = "";
if (preg_match('/<div[^>]*>(.*?)<\/div>/s', $postHtml, $textMatch)) {
    $text = trim(strip_tags(html_entity_decode($textMatch[1])));
}

/********************
 * منع التكرار
 ********************/
$currentHash = sha1($postUrl . $text);
$lastHash = getLastHash($hashFile);

if ($currentHash === $lastHash) {
    // لا يوجد جديد
    exit;
}

saveHash($hashFile, $currentHash);

/********************
 * إرسال إشعار تلجرام
 ********************/
$message  = "🆕 منشور جديد من صفحة العربية\n\n";
if ($text) {
    $message .= "📝 النص:\n" . mb_substr($text, 0, 1500) . "\n\n";
}
$message .= "🔗 الرابط:\n" . $postUrl;

sendTelegram($telegramBotToken, $telegramChatId, $message);

File: fb_alarabiya_bot.py
import requests, re, hashlib, os, time, threading
from html import unescape
from flask import Flask

app = Flask(__name__)

FACEBOOK_PAGE = "https://www.facebook.com/AlArabiya"
TELEGRAM_BOT_TOKEN = "7522002533:AAEQzquyk1AOV71gtyljXeMHfCBJyKv3iE0"
TELEGRAM_CHAT_ID = "5442141079"
HASH_FILE = "last_post_hash.txt"
CHECK_INTERVAL = 300

HEADERS = {"User-Agent": "Mozilla/5.0"}

def fetch_page(url):
    try:
        r = requests.get(url, headers=HEADERS, timeout=20)
        if r.status_code == 200:
            return r.text
    except:
        pass
    return None

def send_telegram(msg):
    url = f"https://api.telegram.org/bot{TELEGRAM_BOT_TOKEN}/sendMessage"
    requests.post(url, data={"chat_id": TELEGRAM_CHAT_ID, "text": msg})

def get_last_hash():
    return open(HASH_FILE).read().strip() if os.path.exists(HASH_FILE) else ""

def save_hash(h):
    open(HASH_FILE, "w").write(h)

def monitor():
    while True:
        html = fetch_page(FACEBOOK_PAGE)
        if html:
            m = re.search(r'/story\.php\?story_fbid=(\d+)&id=(\d+)', html)
            if m:
                post_url = f"https://www.facebook.com/story.php?story_fbid={m.group(1)}&id={m.group(2)}"
                post_html = fetch_page("https://mbasic.facebook.com" + m.group(0))
                text = ""
                if post_html:
                    t = re.search(r'<div[^>]*>(.*?)</div>', post_html, re.S)
                    if t:
                        text = unescape(re.sub(r'<[^>]+>', '', t.group(1))).strip()

                h = hashlib.sha1((post_url + text).encode()).hexdigest()
                if h != get_last_hash():
                    save_hash(h)
                    send_telegram(f"🆕 منشور جديد\n\n{text[:1000]}\n\n{post_url}")
        time.sleep(CHECK_INTERVAL)

@app.route("/")
def home():
    return "Bot is running"

if __name__ == "__main__":
    threading.Thread(target=monitor, daemon=True).start()
    app.run(host="0.0.0.0", port=8080)
